webapp/components/generic/SearchPost/SearchPost.spec.js

```diff
@@ -0,0 +1,64 @@
+import { shallowMount } from '@vue/test-utils'
+import SearchPost from './SearchPost.vue'
+
+const localVue = global.localVue
+localVue.filter('dateTime', d => d)
+
+describe('SearchPost.vue', () => {
+  let mocks, wrapper, propsData
+  beforeEach(() => {
+    mocks = {
+      $t: jest.fn(string => string),
+    }
+    propsData = {
+      option: {
+        title: 'Post Title',
+        commentsCount: 3,
+        shoutedCount: 6,
+        createdAt: '23.08.2019',
+        author: {
+          name: 'Post Author',
+        },
+      },
+    }
+    wrapper = Wrapper()
+  })
+
+  const Wrapper = () => {
+    return shallowMount(SearchPost, { mocks, localVue, propsData })
+  }
+
+  describe('shallowMount', () => {
+    it('renders post title', () => {
+      expect(wrapper.find('.search-option-label').text()).toMatch('Post Title')
+    })
+
+    it('renders post commentsCount', () => {
+      expect(
+        wrapper
+          .find('.search-post-meta')
+          .findAll('span')
+          .filter(item => item.text() === '3')
+          .exists(),
+      ).toBe(true)
+    })
+
+    it('renders post shoutedCount', () => {
+      expect(
+        wrapper
+          .find('.search-post-meta')
+          .findAll('span')
+          .filter(item => item.text() === '6')
+          .exists(),
+      ).toBe(true)
+    })
+
+    it('renders post author', () => {
+      expect(wrapper.find('.search-post-author').text()).toContain('Post Author')
+    })
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-01-07T11:11:31Z" title="Tuesday, January 7th 2020, 12:11:31 pm +01:00">Jan 7, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,62 @@
+import { shallowMount } from '@vue/test-utils'
+import SearchPost from './SearchPost.vue'
+
+const localVue = global.localVue
+localVue.filter('dateTime', d => d)
+
+describe('SearchPost.vue', () => {
+  let mocks, wrapper, propsData
+  beforeEach(() => {
+    mocks = {
+      $t: jest.fn(string => string),
+    }
+    propsData = {
+      option: {
+        title: 'Post Title',
+        commentsCount: 3,
+        shoutedCount: 6,
+        createdAt: '23.08.2019',
+        author: {
+          name: 'Post Author',
+        },
+      },
+    }
+    wrapper = Wrapper()
+  })
+
+  const Wrapper = () => {
+    return shallowMount(SearchPost, { mocks, localVue, propsData })
+  }
+
+  describe('mount', () => {
+    it('renders post title', () => {
+      expect(wrapper.find('.search-option-label').text()).toMatch('Post Title')
+    })
+    it('renders post commentsCount', () => {
+      expect(
+        wrapper
+          .find('.search-post-meta')
+          .findAll('span')
+          .at(0)
+          .text(),
+      ).toMatch('3')
+    })
+    it('renders post shoutedCount', () => {
+      expect(
+        wrapper
+          .find('.search-post-meta')
+          .findAll('span')
+          .at(1)
+          .text(),
+      ).toMatch('6')
+    })
+    it('renders post author', () => {
+      expect(
+        wrapper
+          .find('.search-post-author')
+          .text()
+          .replace(/\s+-\s+/, ' '),
```


why is this necessary?